lib/src/vs/conv.rs

```diff
@@ -0,0 +1,107 @@
+// u64_to_versionstamp converts a u64 to a 10-byte versionstamp
+// assuming big-endian and the the last two bytes are zero.
+pub fn u64_to_versionstamp(v: u64) -> [u8; 10] {
+	let mut buf = [0; 10];
+	buf[0] = (v >> 56) as u8;
+	buf[1] = (v >> 48) as u8;
+	buf[2] = (v >> 40) as u8;
+	buf[3] = (v >> 32) as u8;
+	buf[4] = (v >> 24) as u8;
+	buf[5] = (v >> 16) as u8;
+	buf[6] = (v >> 8) as u8;
+	buf[7] = v as u8;
+	buf
+}
+
+#[allow(unused)]
+// u64_u16_to_versionstamp converts a u64 and a u16 to a 10-byte versionstamp
+// assuming big-endian.
+pub fn u64_u16_to_versionstamp(v: u64, v2: u16) -> [u8; 10] {
+	let mut buf = [0; 10];
+	buf[0] = (v >> 56) as u8;
+	buf[1] = (v >> 48) as u8;
+	buf[2] = (v >> 40) as u8;
+	buf[3] = (v >> 32) as u8;
+	buf[4] = (v >> 24) as u8;
+	buf[5] = (v >> 16) as u8;
+	buf[6] = (v >> 8) as u8;
+	buf[7] = v as u8;
+	buf[8] = (v2 >> 8) as u8;
+	buf[9] = v2 as u8;
+	buf
+}
+
+#[allow(unused)]
+// u64_u16_to_versionstamp converts a u64 and a u16 to a 10-byte versionstamp
+// assuming big-endian.
+pub fn u16_u64_to_versionstamp(v: u16, v2: u64) -> [u8; 10] {
+	let mut buf = [0; 10];
+	buf[0] = (v >> 8) as u8;
+	buf[1] = v as u8;
+	buf[2] = (v2 >> 56) as u8;
+	buf[3] = (v2 >> 48) as u8;
+	buf[4] = (v2 >> 40) as u8;
+	buf[5] = (v2 >> 32) as u8;
+	buf[6] = (v2 >> 24) as u8;
+	buf[7] = (v2 >> 16) as u8;
+	buf[8] = (v2 >> 8) as u8;
+	buf[9] = v2 as u8;
+	buf
+}
+
+// u128_to_versionstamp converts a u128 to a 10-byte versionstamp
+// assuming big-endian.
+#[allow(unused)]
+pub fn u128_to_versionstamp(v: u128) -> [u8; 10] {
+	let mut buf = [0; 10];
+	buf[0] = (v >> 72) as u8;
+	buf[1] = (v >> 64) as u8;
+	buf[2] = (v >> 56) as u8;
+	buf[3] = (v >> 48) as u8;
+	buf[4] = (v >> 40) as u8;
+	buf[5] = (v >> 32) as u8;
+	buf[6] = (v >> 24) as u8;
+	buf[7] = (v >> 16) as u8;
+	buf[8] = (v >> 8) as u8;
+	buf[9] = v as u8;
+	buf
```

Comment:
Shall we manage overflow?

Reply:
Good idea!! Does f8a7eef0 look good to you?